lib/src/scrollbar95.dart

```diff
@@ -0,0 +1,383 @@
+    double trackWidth = trackRenderBox.size.width;
+    double trackLeft = trackRenderBox.localToGlobal(Offset.zero).dx;
+    double thumbPosition = details.globalPosition.dx - trackLeft;
+
+    double thumbPositionNormalised = thumbPosition / trackWidth;
+
+    double nextPosition =
+        thumbPositionNormalised * (_controller!.position.maxScrollExtent);
+
+    nextPosition =
+        clampDouble(nextPosition, 0, _controller!.position.extentTotal);
+
+    _controller!.jumpTo(nextPosition);
+  }
+
+  void _findControllerAndListen(BuildContext context) {
+    if (_controller != null) return;
+
+    // Find the scroll controller and attach the listener.
+    if (widget.child is ScrollView &&
+        (widget.child as ScrollView).controller != null) {
+      _controller = (widget.child as ScrollView).controller;
+    } else if (widget.child is SingleChildScrollView &&
+        (widget.child as SingleChildScrollView).controller != null) {
+      _controller = (widget.child as SingleChildScrollView).controller;
+    } else {
+      _controller = PrimaryScrollController.of(context);
+    }
+
+    _controller?.addListener(_positionListener);
+
+    // Call the listener once to set the initial position.
+    WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
+      _scrollDirection = _controller!.position.axis;
+      _positionListener();
+      _getThumbExtent();
+    });
+  }
+
+  void _getThumbExtent() {
+    ScrollMetrics scrollMetrics = _controller!.position;
+
+    final double totalContentExtent = scrollMetrics.extentTotal;
+
+    // Thumb extent reflects fraction of content visible, as long as this
+    final double fractionVisible = clampDouble(
+      (scrollMetrics.extentInside) / totalContentExtent,
+      0.0,
+      1.0,
+    );
+
+    final double traversableTrackExtent =
+        scrollMetrics.viewportDimension - (2 * widget.scrollbarThickness);
+
+    final double thumbExtent = max(
+      min(
+        traversableTrackExtent * fractionVisible,
+        traversableTrackExtent,
+      ),
+      traversableTrackExtent * fractionVisible,
+    );
+
+    final double fractionOverscrolled =
+        1.0 - scrollMetrics.extentInside / scrollMetrics.viewportDimension;
+
+    final double minLength = widget.minThumbHeight;
+
+    final double safeMinLength = min(minLength, traversableTrackExtent);
+    final double newMinLength =
+        (scrollMetrics.extentBefore > 0 && scrollMetrics.extentAfter > 0)
+            ? safeMinLength
+            : safeMinLength *
+                (1.0 - clampDouble(fractionOverscrolled, 0.0, 0.2) / 0.2);
+
+    _thumbExtent =
+        clampDouble(thumbExtent, newMinLength, traversableTrackExtent);
+  }
+
+  @override
+  void didUpdateWidget(covariant Scrollbar95 oldWidget) {
+    // Make controller null so on next rebuild, it is re-fetched.
+    _controller = null;
+    super.didUpdateWidget(oldWidget);
+  }
+}
+
+class _CheckeredBackgroundPainter extends CustomPainter {
```

Comment:
This is so cool, I think it could be made available openly within the library to allow using the checkered background in more places.